File: src/main/python/leetcode/array/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for (idx, num) in enumerate(nums):
            if num in dict:
                return [idx, dict[num]]
            else:
                dict[target - num] = idx
        return [-1, -1]
        

File: src/main/go/array/two_sum.go
package array

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for i, num := range nums {
		if _, ok := m[num]; ok {
			//do something here
			return []int{i, m[num]}
		} else {
			m[target-num] = i
		}
	}

	return []int{-1, -1}
}
